osquery/sql/virtual_table.cpp

```diff
@@ -19,8 +19,8 @@ namespace sqlite {
 int xOpen(sqlite3_vtab *p, sqlite3_vtab_cursor **ppCursor) {
   int rc = SQLITE_NOMEM;
   BaseCursor *pCur = new BaseCursor;
-  if (pCur) {
-    memset(pCur, 0, sizeof(BaseCursor));
```

Comment:
A `memset` here will overwrite the included `std::vector`/`QueryData`. All we need to do is initialize pointers and integrals.